test/services/challenges_repository.js

```diff
@@ -9,49 +9,191 @@ This Source Code Form is “Incompatible With Secondary Licenses”, as defined
+        {challengeId: '-1'},
+        {challengeId: '7'},
+        {challengeId: '7'}
+      ];
+
+      it('should remove single challenges that are resolved or timed out', async () => {
+        expect(challengesRepository.filterOutFinished(allChallenges, resolvedChallenges, timedOutChallenges)).to.deep.equal([
+          {challengeId: '4', count: 1},
+          {challengeId: '5', count: 1}
+        ]);
+      });
+    });
+
+    describe('Multiple challenges', () => {
+      const allChallenges = [
+        {challengeId: '1', count: 2},
+        {challengeId: '2', count: 1},
+        {challengeId: '3', count: 4}
+      ];
+      const resolvedChallenges = [
+        {challengeId: '1'},
+        {challengeId: '1'},
+        {challengeId: '2'},
+        {challengeId: '3'},
+        {challengeId: '3'}
+      ];
+      const timedOutChallenges = [{challengeId: '3'}];
+
+      it('should remove challenges only if resolution count is not less then creation count', async () => {
+        expect(challengesRepository.filterOutFinished(allChallenges, resolvedChallenges, [])).to.deep.equal([
+          {challengeId: '3', count: 4}
+        ]);
+      });
+
+      it('should remove challenge when timed out', async () => {
+        expect(challengesRepository.filterOutFinished(allChallenges, [], timedOutChallenges)).to.deep.equal([
+          {challengeId: '1', count: 2},
+          {challengeId: '2', count: 1}
+        ]);
+      });
+
+      it('timed out and resolved work together', async () => {
+        expect(challengesRepository.filterOutFinished(allChallenges, resolvedChallenges, timedOutChallenges)).to.deep.equal([]);
+      });
+    });
+  });
+
+  describe('extractChallengeFromEvent', () => {
+    const sheltererId = 1;
+    const bundleId = 2;
+    const challengeId = 3;
+    const count = 1;
+    const events = [
+      {
+        returnValues: {
+          sheltererId,
+          bundleId,
+          challengeId,
+          count
+        }
+      },
+      {
+        returnValues: {
+          sheltererId,
+          bundleId,
+          challengeId: 100,
+          count
+        }
+      },
+      {
+        returnValues: {
+          sheltererId,
+          bundleId,
+          challengeId,
+          count
+        }
+      }];
+    const challenges = [
+      {sheltererId, bundleId, challengeId, count},
+      {sheltererId, bundleId, challengeId: 100, count},
+      {sheltererId, bundleId, challengeId, count}];
+
+    beforeEach(() => {
+      challengesRepository = new ChallengesRepository();
+    });
+
+    it('extracts challengeId, sheltererId, bundleId from events list', async () => {
+      expect(challengesRepository.extractChallengeFromEvent(events)).to.deep.equal(challenges);
+    });
+  });
+
   describe('ongoingChallenges', () => {
     const sheltererId = 1;
     const bundleId = 2;
     const challengeId = 3;
     const fromBlock = 4;
+    const count = 1;
     const challengeDuration = 5;
     const events = [
```

Comment:
IMO providing this extensive data for this kind of tests is unnecessary. The tests below check if `ongoingChallenges` delegates it's work properly. This placeholders could/should be as small as possible (as to not to introduce noise), just enough to be identified.